client/src/App.tsx

```
@@ -0,0 +1,192 @@
+import React, { useEffect, useState } from 'react';
+
+type Message = {
+  author: string;
+  text: string;
+  time: string;
+};
+
+export function App() {
+  const [messages, setMessages] = useState<Message[]>([]);
+  const [text, setText] = useState('');
+  const [rooms, setRooms] = useState<string[]>([]);
+  const [currentRoom, setCurrentRoom] = useState('general');
+  const [newRoom, setNewRoom] = useState('');
+
+  useEffect(() => {
+    fetch('http://localhost:3000/rooms')
+      .then((res) => res.json())
+      .then((data) => setRooms(data));
+  }, []);
+
+  useEffect(() => {
+    setMessages([]);
+
+    const fetchMessages = () => {
+      fetch(`http://localhost:3000/rooms/${currentRoom}/messages`)
+        .then((res) => res.json())
+        .then((data) => setMessages(data));
+    };
+
+    fetchMessages();
+    const timer = setInterval(fetchMessages, 1000);
+
+    return () => clearInterval(timer);
+  }, [currentRoom]);
+
+  useEffect(() => {
+    let saved = localStorage.getItem('username');
+    if (!saved) {
+      saved = prompt('Enter a username') || 'Anonymous';
+      localStorage.setItem('username', saved);
+    }
+  }, []);
+
+  async function sendMessage(e: React.FormEvent) {
+    e.preventDefault();
+    const username = localStorage.getItem('username') || 'Anonymous';
```

sendMessage currently falls back to `'Anonymous'` when localStorage lacks a username. The requirements state the client must not allow sending a message without a current username and selected room (req. 3.14). Instead of silently defaulting, block sending and prompt the user to set a username (or open the username UI). Also validate `currentRoom` is a valid room before sending.